### DISCUSSION TITLE
How do I find users with high contributions?

### DISCUSSION THREAD

**Discussion by rhysluxford** [community/community#138975] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I am currently looking to find specific engineers with very high contributions over the years. 

Is there a way to search on Githubs "advanced search" tab, where I can select users with a high amount of contributions?

If it's not possible, are there any other alternative ways?

**Comment by LiteBrite82** [comment]
@rhysluxford 
Is this search constrained to a specific org, or are you looking for all of github?

**Comment by dusova** [comment, marked as answer]
GitHub’s advanced search doesn’t provide a direct filter specifically for user contributions. However, you can use a combination of search features and other methods to identify highly active contributors:

1. **Search for Repositories**: You can search for repositories using keywords, then sort the results by the number of stars or forks. This often leads to repositories with high contributions. For example, you can use:
   ```
   stars:>1000 language:JavaScript
   ```
   Adjust the language and stars according to your needs.

2. **Explore Contributors**: Once you find a repository, you can go to the "Insights" tab and then click on "Contributors." This section lists contributors by the number of commits, which can help you identify highly active engineers.

3. **Use GitHub API**: If you're comfortable with programming, you can leverage the GitHub API to fetch contribution data. You can retrieve user profiles and their contributions programmatically.

4. **GitHub Search Operators**: Use specific search operators in the GitHub search bar, like:
   ```
   users:username
   ```
   This will bring up a user's repositories, and you can see their contributions from there.

5. **Third-Party Tools**: Consider using third-party analytics tools like **GitHub Awards** or **GitHub Stars** that can help identify top contributors based on various metrics.

These methods should help you find engineers with high contributions effectively!

**Reply by rhysluxford** [reply]
Thank you @madtethys this is very helpful. :)

**Reply by dusova** [reply]
You're welcome. I'd be very happy if you marked it as a solution.

**Reply by rhysluxford** [reply]
Done :)

**Reply by dusova** [reply]
Thanks :)

**Comment by tomguluson92** [comment]
So how about ppl who wrote code frequently on their private repos, green dots much, but less open source, can we track that?

**Comment by shekhfaisal2110** [comment]
Currently, GitHub’s built-in Advanced Search does not provide a direct filter to find users by their total number of contributions (commits, pull requests, issues, etc.). The “Users” search allows you to filter by location, language, or number of repositories, but not by contribution activity or commit history.

However, there are a few alternative approaches you can use:

1.GitHub’s GraphQL API (v4)
You can query contribution data for users programmatically.
The contributionsCollection object allows you to retrieve metrics like commit counts, pull requests, issue activity, and more.

Example query:
{
  user(login: "username") {
    contributionsCollection {
      contributionCalendar {
        totalContributions
      }
    }
  }
}


This approach is ideal if you want to build custom leaderboards or identify high-contributing developers.

2.Third-Party Tools
Platforms like GitHub Wrapped, Commits.top, or GitAwards aggregate public contribution data and rank users by their activity.
These tools use GitHub’s public API to identify users with high contribution counts by region or language.

3.Organization Insights (for org owners)
If you manage a GitHub Organization, you can use Insights → Contributors to view top contributors over time for your repositories.
This view helps identify engineers with consistent contributions within your organization’s projects.

**Reply by tomguluson92** [reply]
Thanks!

**Comment by EthanUnlocked690** [comment]
I'm looking for a contributor or 2 to help with my brawl stars Leon wiki.
